messages/proposed_event.go

```diff
@@ -1,14 +1,108 @@
 package messages
 
 import (
+	"encoding/json"
+
 	uuid "github.com/gofrs/uuid"
 )
 
 // ProposedEvent ...
 type ProposedEvent struct {
-	EventID      uuid.UUID
-	EventType    string
-	ContentType  string
-	Data         []byte
-	UserMetadata []byte
+	eventID      uuid.UUID
+	eventType    string
+	contentType  string
+	data         []byte
+	metadata []byte
+}
+
+func NewJsonProposedEvent(eventType string, payload interface{}) (ProposedEvent, error) {
+	event := ProposedEvent {}
+	bytes, err := json.Marshal(payload)
+
+	if err != nil {
+		return event, err
+	}
+
+	event.eventType = eventType
+	event.contentType = "application/json"
+	event.data = bytes
+	event.metadata = []byte{}
+
+	return event, nil
+}
+
+func NewBinaryProposedEvent(eventType string, bytes []byte) ProposedEvent {
+	event := ProposedEvent {
+		eventType: eventType,
+		contentType: "application/octet-stream",
+		data: bytes,
+		metadata: []byte{},
+	}
+
+	return event
+}
+
+func (event ProposedEvent) GetEventID() uuid.UUID {
+	return event.eventID
+}
+
+func (event ProposedEvent) GetEventType() string {
+	return event.eventType
+}
+
+func (event ProposedEvent) GetContentType() string {
+	return event.contentType
+}
+
+func (event ProposedEvent) GetData() []byte {
+	return event.data
+}
+
+func (event ProposedEvent) GetMetadata() []byte {
+	return event.metadata
+}
+
+func (event ProposedEvent) EventID(value uuid.UUID) ProposedEvent {
+	event.eventID = value
+	return event
+}
+
+func (event ProposedEvent) EventType(value string) ProposedEvent {
+	event.eventType = value
+	return event
+}
+
+func (event ProposedEvent) ContentType(value string) ProposedEvent {
+	event.contentType = value
+	return event
+}
+
+func (event ProposedEvent) JsonData(payload interface{}) (ProposedEvent, error) {
```

Comment:
I don't see that code used anywhere. Isn't it actually a duplication of `NewJsonProposedEvent` in some way. As a consumer, I'd be confused about what to use.

IMO it would be better to just leave `NewJsonProposedEvent`.